docs/scipp_widgets_intro.ipynb

```diff
@@ -0,0 +1,179 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import scipp as sc\n",
+    "d = sc.Dataset(\n",
+    "    {\n",
+    "    'alice': sc.Variable(['z', 'y', 'x'], values=np.random.rand(10, 10, 10),\n",
+    "                         variances=0.1*np.random.rand(10, 10, 10)),\n",
+    "    'bob': sc.Variable(['x', 'z'], values=np.arange(0.0, 10.0, 0.1).reshape(10, 10),\n",
+    "                       variances=0.1*np.random.rand(10, 10))\n",
+    "    },\n",
+    "    coords={\n",
+    "        'x': sc.Variable(['x'], values=np.arange(11.0), unit=sc.units.m),\n",
+    "        'y': sc.Variable(['y'], values=np.arange(11.0), unit=sc.units.m),\n",
+    "        'z': sc.Variable(['z'], values=np.arange(11.0), unit=sc.units.m)})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Wrapping a display function\n",
+    "\n",
+    "scipp-widgets allows functions to be wrapped in simple graphical interfaces. \n",
+    "As a first example, take the `scipp.plot.plot` function. This takes a scipp object and plots it. \n",
+    "In its simplest form, this only has one input; the object to be plotted. \n",
+    "\n",
+    "We will need to create an input specification object for this describing how we want it displayed, what validation we wish to perform and which argument of the underlying function it should correspond to.\n",
+    "\n",
+    "The minimum we need here is the name of the underlying function argument."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scipp_widgets.input_spec import InputSpec\n",
+    "from scipp_widgets.widgets import DisplayWidget\n",
+    "data_input = InputSpec('scipp_obj', eval_input=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `eval_input` allows variables in the notebooks global scope to be passed to this input by name.\n",
+    "\n",
+    "A widget which wraps the plot function can then be constructed like so:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DisplayWidget(sc.plot.plot, 'Plot', [data_input])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here the arguments are: function to wrap, name inside the button which triggers the execution, list of required input objects."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Wrapping a processing function\n",
+    "Another common case is to wrap a function which processes an input and produces an output. \n",
+    "This can be achieved using the `ProcessWidget` class. \n",
+    "As an example, we construct a wrapper around the `scipp.sum` function below."
```

Comment:
This seems like a simpler example than the plotting, should it come first?

Reply:
My thinking here was that programmatically the Display widget is simpler as it doesn't need to worry about putting outputs back in the global scope. That doesn't matter to the user though so if the other way around makes more sense from a conceptional standpoint(which I think it does) I'll switch then around.